packages/lib/timezone.ts

```diff
@@ -25,29 +25,37 @@ export const filterByCities = (tz: string, data: ICity[]): ICity[] => {
 };
 
 export const addCitiesToDropdown = (cities: ICity[]) => {
-  const cityTimezones = cities?.reduce((acc: { [key: string]: string }, city: ICity) => {
+  const cityTimezones: { [key: string]: string } = {};
+
+  cities?.forEach((city: ICity) => {
     if (city.timezone !== null && !isProblematicTimezone(city.timezone)) {
-      acc[city.timezone] = city.city;
+      cityTimezones[city.timezone] = city.city;
     }
-    return acc;
-  }, {});
+  });
 
   additionalTimezones.forEach((tz) => {
-    cityTimezones[tz.identifier] = tz.name;
+    cityTimezones[tz.city] = tz.timezone;
```

Comment:
I think by changing this line to "cityTimezones[tz.timezone] = tz.city;" may solve the issue you mentioned  "in the cityTimezones logic the city would be "Eastern European Time" and the "timezone": Europe/Bucharest",  I just took a glance at the code and thought this might work, havent coded or tested, what do you think about it. @emrysal